pins/drivers.py

```diff
@@ -15,14 +16,38 @@
 
 
 def _assert_is_pandas_df(x):
-    import pandas as pd
+    df_family = _get_df_family(x)
```

Comment:
Is there a benefit to using `_get_df_family()` over checking for a pandas dataframe?

Reply:
The motivation is to centralize the logic for deciding a DataFrame's type into one single location.

I think my preference for consistency's sake is to refactor the `_assert_is_pandas_df` function away, and use a similar strategy of case checking as in these lines:

https://github.com/nathanjmcdougall/pins-python/blob/2c2e5029c300631822d1326d5b94cde895b4294b/pins/drivers.py#L200-L211

The disadvantage is that the error messages could get a bit repetitive.

Reply:
I agree with eventually refactoring `_assert_is_pandas_df` out in favor of `_get_df_family`! I think we can streamline the errors a bit 😄